src/vm/arm64/asmhelpers.S

```diff
@@ -310,29 +310,46 @@ LOCAL_LABEL(shadowupdateend):
     ldp  x12, x13, [sp],#16
 #endif
 
-    // Branch to Exit if the reference is not in the Gen0 heap
+    // Branch to DoneCardTable if the reference is not in the Gen0 heap
```

Comment:
> Since there are no barriers, between the card table and the WW, there is no guarantee which order they are observable by another thread 

The GC tends to use FlushWriteBuffers API that makes the order guaranteed from another threads. I do not know whether it is the case here ... just making a general point that common sense rules may not apply.